src/main/java/school/hei/haapi/endpoint/rest/controller/MpbsVerificationController.java

```diff
@@ -28,6 +28,10 @@ public List<MpbsVerification> getMpbsVerifications(
 
   @GetMapping(value = "/verify-transactions")
   public List<MpbsVerification> verifyTransactions() {
+    if (mpbsVerificationService.checkMobilePaymentThenSaveVerification() == null) {
+      return List.of();
+    }
```

Comment:
why do you execute "checkMobilePaymentThenSaveVerification" multiple times in a row instead of just once?